provisioning/transport/http/devdoc/http_requirements.md

```diff
@@ -0,0 +1,89 @@
+  register(authorization?: string, forceRegistration?: boolean, done(err: Error, result?: Any) => void) : void;
+}
+```
+
+## Public Interface
+
+### constructor(idScope: string, registrationId: string) : void
+The `constructor` creates an Http transport object used to communicate with the Azure device provisioning service
+
+**SRS_NODE_PROVISIONING_HTTP_18_001: [** The `Http` constructor shall accept the following properties:
+- `idScope` - a string specifiying the scope of the provisioning operations,
+- `registrationId` - the registration id for the specific device 
+**]**
+
+
+### register(authorization?: string, forceRegistration?: boolean, callback (err: Error, result?: Any) => void) : void;
+`register` calls into the service to register the given device with the provisioning service.
+
+**SRS_NODE_PROVISIONING_HTTP_18_002: [** `register` shall throw a `ReferenceError` if `callback` is falsy. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_004: [** The registration request shall use Http 1.1. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_005: [** The registration request shall include the current `api-version` as a URL query string value. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_006: [** The registration request shall specify the following in the Http header:
+  Accept: application/json
+  Content-Type: application/json; charset=utf-8 **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_007: [** If an `authorization` string is specifed, the registration request shall include this in the Http header. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_008: [** If `forceRegistration` is specified, the registration request shall include this as a query string value. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_009: [** `register` shall shall POST the registration request to 'https://global.azure-devices-provisioning.net/{idScope}/registrations/{registrationId}' **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_010: [** `register` shall wait for a registration response. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_011: [** If the registration request times out, `register` shall call the `callback` with `TimeoutError` error **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_012: [** If the registration response contains a body, `register` shall deserialize this into a JSON object. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_013: [** If registration response body fails to deserialize, `register` will call the `callback` with a `SyntaxError` error. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_014: [** If the registration response has a failed status code, `register` shall use `translateError` to translate this to a common error object and pass this into the `callback` function along with the deserialized body of the response. **]**
+
+**SRS_NODE_PROVISIONING_HTTP_18_015: [** If the registration response has a success code with a 'status' of 'Assigned', `register` call the `callback` with `err` == `null` and result `containing` the deserialized body **]**
```

Comment:
>`err` == `null` [](start = 156, length = 15)

I've learned recently that it's recommended to avoid using null when undefined can do. I'm torn between starting to use undefined and make this inconsistent with the rest of the SDK or keeping consistency (and to a certain extent, a bad practice) alive. not sure what to suggest but letting you know what's going on.